src/blocks/weather.rs

```diff
@@ -105,7 +107,7 @@ pub mod open_weather_map;
 
 const IP_API_URL: &str = "https://ipapi.co/json";
 
-static LAST_AUTOLOCATE: Mutex<Option<AutolocateResult>> = Mutex::new(None);
+static LAST_AUTOLOCATE: Lazy<Mutex<Option<AutolocateResult>>> = Lazy::new(|| Mutex::new(None));
```

Comment:
Why this change? This mutex is not locked across `.await`s. In cases like this [tokio advises to use ordinary mutexes](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use).

Reply:
I did that because of the use super::* in own, but can use the two mutex types with an alias or qualify wich mutex to use

Reply:
If it is possible to get rid of `LOCATION_CACHE` then this will resolve itself, I guess.